extensions/amp-experiment/1.0/test/test-amp-experiment.js

```diff
@@ -147,17 +158,45 @@ describes.realWin('amp-experiment', {
     });
   });
 
-  it('should apply the mutations from the variant', () => {
+  it('should throw if the chosen experiment / ' +
+    'variant config has too many mutations', () => {
+
+    const tooManyMutationsConfig = {
+      'experiment-1': {
+        variants: {
+          'variant-a': {
+            weight: 50,
+            mutations: (new Array(200)).fill({}),
+          },
+          'variant-b': {
+            weight: 50,
+            mutations: (new Array(200)).fill({}),
+          },
+        },
+      },
+    };
+
+    addConfigElement(
+        'script',
+        'application/json',
+        JSON.stringify(tooManyMutationsConfig)
+    );
+    stubAllocateVariant(sandbox, tooManyMutationsConfig);
+
+    expectAsyncConsoleError(/Max number of mutations/);
+    return experiment.buildCallback().then(() => {
+      throw new Error('must have failed');
+    }, e => {
+      expect(e).to.match(/Max number of mutations/);
+    });
+  });
+
+  it('should match the variant to the experiment', () => {
     addConfigElement('script');
-    const stub = sandbox.stub(variant, 'allocateVariant');
-    stub.withArgs(ampdoc, 'experiment-1', config['experiment-1'])
-        .returns(Promise.resolve('variant-a'));
-    stub.withArgs(ampdoc, 'experiment-2', config['experiment-2'])
-        .returns(Promise.resolve('variant-d'));
-    stub.withArgs(ampdoc, 'experiment-3', config['experiment-3'])
-        .returns(Promise.resolve(null));
+    stubAllocateVariant(sandbox, config);
 
     const applyStub = sandbox.stub(experiment, 'applyMutations_');
+    sandbox.stub(experiment, 'validateExperimentToVariant_');
```

Comment:
since the mutation number doesn't exceed 70, can we not stub the method?

Reply:
Gonna keep the stub, so we can isolate the code to just running amp-experiment.js 😄

Reply:
Remove this, and if still works, keep it removed. If it doesn't re-add it 😂

Reply:
Tested this, and totally need this or else it fails because my mock config doesn't contain actual mutations (Which is what we want) 😄